src/broker/broker.c

```diff
@@ -1246,12 +1211,22 @@ static int load_module (broker_ctx_t *ctx,
             errno = EINVAL;
             return -1;
         }
-        if (search_file_ext (path, ".so", searchpath, &fullpath) < 0) {
+        if (asprintf (&pattern, "%s.so*", path) < 0) {
+            errprintf (error, "out of memory");
+            return -1;
+        }
+        if (!(files = dirwalk_find (searchpath,
+                                    DIRWALK_REALPATH,
+                                    pattern,
+                                    1,
+                                    NULL,
+                                    NULL))
+            || zlist_size (files) == 0) {
             errprintf (error, "module not found in search path");
             errno = ENOENT;
-            return -1;
+            goto error;
         }
-        path = fullpath;
+        path = zlist_first (files);
```

Comment:
I could see a developer messing up and having multiple modules of the same name (name.so and name.so.1.0.0) in the same dir and not realizing it.  Would it worthwhile to dump/log some warning just in case?  Something simple like "multiiple name.so found, using <path>" or something?   I know the odds are low and unlikely ... just sort of thinking out loud.

Reply:
These files are normally generated by build systems so I would think the common case would be just  `foo.so` or `foo.so.0.0.0` and possibly some additional symlinks pointing to the same place.  If there is confusion, `flux module list -l` can show the full path to the DSO that was loaded.  Maybe enough?

Reply:
if `flux module list -l` shows the full path, then that seems like more than enough!